hcpannot/analysis/lmm.py

```diff
@@ -0,0 +1,222 @@
+
+    # get the variance explained by researchers and store the data in the dictionary
+    contour_data['lh']['rater'] = lh_dat['varex_rater'].values
+    contour_data['rh']['rater'] = rh_dat['varex_rater'].values
+    
+    # get the residual of x coordinates and store the data in the dictionary
+    contour_data['lh']['residual'] = lh_dat['residual'].values
+    contour_data['rh']['residual'] = rh_dat['residual'].values
+    
+    # check if r2s_variance was a column in the dataframe
+    if 'r2s_variance' in result.columns:
+        # get the variance of the random slopes and store the data in the dictionary
+        contour_data['lh']['r2s_variance'] = lh_dat['r2s_variance'].values
+        contour_data['rh']['r2s_variance'] = rh_dat['r2s_variance'].values
+    
+    return contour_data
+
+def lwplot(x, y, axes=None, fill=True, edgecolor=None, color=None, **kw):
+    '''
+    lwplot(x, y) is equivalent to pyplot.plot(x, y), however the linewidth or lw options
+      are interpreted in terms of the the coordinate system instead of printer points.
+    lwplot(x, y, ax) plots on the given axes ax.
+    
+    All optional arguments that can be passed to pyplot's Polygon can be passed to lwplot.
+    '''
+    from neuropythy.util import zinv
+    lw = kw['linewidth'] if 'linewidth' in kw else kw['lw'] if 'lw' in kw else None
+    if 'linewidth' in kw: lw = kw.pop('linewidth')
+    elif 'lw' in kw: lw = kw.pop('lw')
+    else: lw = 0
+    axes = plt.gca() if axes is None else axes
+    if len(x) < 2: raise ValueError('lwplot line must be at least 2 points long')
+    # we plot a particular thickness; we need to know the orthogonals at each point...
+    pts = np.transpose([x,y])
+    dd  = np.vstack([[pts[1] - pts[0]], pts[2:] - pts[:-2], [pts[-1] - pts[-2]]])
+    nrm = np.sqrt(np.sum(dd**2, 1))
+    dd  *= np.reshape(zinv(nrm), (-1,1))
+    dd  = np.transpose([dd[:,1], -dd[:,0]])
+    # we make a polygon or a trimesh...
+    if pimms.is_vector(lw): lw = np.reshape(lw, (-1,1))
+    xy = np.vstack([pts + lw*dd, np.flipud(pts - lw*dd)])
+    n = len(pts)
+    if pimms.is_vector(color) and len(color) == n:
+        clr = np.concatenate([color, np.flip(color)])
+        (nf0,nf1) = (np.arange(n-1), np.arange(1,n))
+        (nb0,nb1) = (2*n - nf0 - 1,  2*n - nf1 - 1)
+        tris = np.hstack([(nf0, nf1, nb0), (nb0, nb1, nf1)]).T
+        (x,y) = xy.T
+        tri = mpl.tri.Triangulation(x, y, tris)
+        if 'cmap' not in kw: kw['cmap'] = 'hot'
+        return axes.tripcolor(tri, clr, shading='gouraud',
+                              linewidth=0, **kw)
+    else:
+        pg  = plt.Polygon(xy, closed=True, fill=fill, edgecolor=edgecolor,
+                          linestyle=None, linewidth=0, color=color, **kw)
+        return axes.add_patch(pg)
+    
+def plot_hmap(x, y, hemi, contour, lm_result, ax, cmap='hot', residual=False, max_var=None):
+    
+    '''
+    Plot the heatmap of the variance explained by researchers or subjects
+    '''
```

Comment:
I would add a note in the docstring that this is for plotting the heatmaps of variance explained along the contours specifically.